File: ignite-nest/src/comment/infrastructure/web/comment.controller.ts
import { Controller, Get, Inject, Query } from '@nestjs/common';
import { CommentListUsecase } from '../../domain/usecase/comment-list.usecase';
import { ApiTags } from '@nestjs/swagger';
import {
  COMMENT_CONTROLLER,
  COMMENT_CONTROLLER_GROUP,
  COMMENT_LIST_USECASE,
} from '../../constants/comment.token';
import { GetCommentsSwagger } from './decorator/comment-controller.decorator';
import { PageRequestDto } from '../../../common/dto/page-request.dto';

@ApiTags(COMMENT_CONTROLLER_GROUP)
@Controller(COMMENT_CONTROLLER)
export class CommentController {
  constructor(
    @Inject(COMMENT_LIST_USECASE)
    private readonly commentListUsecase: CommentListUsecase,
  ) {}

  @Get()
  @GetCommentsSwagger()
  async getComments(@Query() request: PageRequestDto) {
    return await this.commentListUsecase.execute(request);
  }
}

File: ignite-spring/src/main/kotlin/org/rookedsysc/ignitespring/comment/infrastructure/web/CommentController.kt
package org.rookedsysc.ignitespring.comment.infrastructure.web

import jakarta.validation.constraints.Size
import org.rookedsysc.ignitespring.comment.domain.usecase.CommentListUsecase
import org.rookedsysc.ignitespring.comment.infrastructure.web.decorator.CommentControllerDecorator
import org.rookedsysc.ignitespring.comment.infrastructure.web.dto.CommentListDto
import org.springframework.data.domain.Pageable
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/comments")
class CommentController(
    private val commentListUsecase: CommentListUsecase
) : CommentControllerDecorator {

    @GetMapping
    override fun getComments(
        @RequestParam(value = "page") page: Int,
        @RequestParam(value = "size") size: Int
    ): List<CommentListDto> {
        val request: Pageable = Pageable.ofSize(size).withPage(page)
        return commentListUsecase.execute(request)
    }
}
